server/src/main/java/org/jboss/as/server/deployment/module/ModuleSpecProcessor.java

```diff
@@ -282,14 +282,9 @@ private ServiceName createModuleService(final DeploymentPhaseContext phaseContex
         sb.setInitialMode(Mode.ON_DEMAND);
         sb.install();
 
-        final List<ModuleDependency> allDependencies = new ArrayList<ModuleDependency>();
-        allDependencies.addAll(dependencies);
-        allDependencies.addAll(localDependencies);
-        allDependencies.addAll(userDependencies);
-
         ModuleResolvePhaseService.installService(phaseContext.getServiceTarget(), moduleDefinition);
 
-        return ModuleLoadService.install(phaseContext.getServiceTarget(), moduleIdentifier, allDependencies);
+        return ModuleLoadService.install(phaseContext.getServiceTarget(), moduleIdentifier, dependencies, localDependencies, userDependencies);
```

Comment:
You are passing `dependencies, localDependencies, userDependencies);` 

The method signature is `install(... ,final List<ModuleDependency> userDependencies, final List<ModuleDependency> localDependencies) {`

So localDependencies and userDependencies should switch places.

Tracked in https://issues.jboss.org/browse/WFCORE-4327

Reply:
Good catch @rsvoboda , yes, you're right, thanks!

Reply:
The fact that the test pass is because the error is corrected later in https://github.com/wildfly/wildfly-core/pull/3661/files/c51e74ff348d0b8febd74068b5ee53ba5117529a#diff-a5f0937871539867ca86d662b4e2fd09R150 where the list are swhitched again